File: Chap5/parity.lua
local bit = require 'numberlua'

local function parity1(x)
  local result = 0
  while (x ~= 0) do
     result = bit.bxor(result, bit.band(x, 1))
     x = bit.rshift(x, 1)
  end
  return result
end

print(parity1(8))

File: Chap5/R/parity.r
require('bitops')

parity1 <- function(x) {
  result <- 0
  while (x != 0) {
    result <- bitXor(result, bitAnd(x, 1))
    x <- bitShiftR(x, 1)
  }
  return(result)
}

print(parity1(8))
print(parity1(9))